File: docs/APIs-and-SDKs/SDK-Documentation/complete-example/flutter/completeExample.dart
import 'package:absmartly_sdk/absmartly_sdk.dart';

void main() async {
  // Initialize the SDK
  final ClientConfig clientConfig = ClientConfig()
    ..setEndpoint("https://your-company.absmartly.io/v1")
    ..setAPIKey("YOUR-API-KEY")
    ..setApplication("website")
    ..setEnvironment("production");

  final ABSmartly sdk = ABSmartly(
    ABSmartlyConfig.create()
        .setClient(Client.create(clientConfig)),
  );

  // Create a context for this user
  final ContextConfig contextConfig = ContextConfig.create()
      .setUnit("user_id", "user-12345");
  final Context context = await sdk
      .createContext(contextConfig)
      .waitUntilReady();

  // Check which variant the user is in
  final int treatment =
      await context.getTreatment("homepage_banner_experiment");

  if (treatment == 1) {
    // Variant B: show a personalized banner
    showBanner("Welcome back, we have new deals for you!");
  } else {
    // Variant A (control): show the existing banner
    showBanner("Welcome back!");
  }

  // Use a variable set in the Web Console for more flexibility
  final String buttonColor =
      await context.getVariableValue("button.color", "blue");
  setButtonColor(buttonColor);

  // Track when the user completes a key action
  context.track("cta_click", {"page": "homepage"});

  // Track a purchase with revenue data
  context.track("purchase", {
    "revenue": 49.99,
    "item_count": 3,
  });

  // Close the context to flush remaining events
  await context.close();
}

File: docs/APIs-and-SDKs/SDK-Documentation/complete-example/vue3/completeExample.js
// main.js - Initialize the SDK as a Vue plugin
import { createApp } from "vue";
import App from "./App.vue";
import absmartly from "@absmartly/vue3-sdk";

const app = createApp(App);

app.use(absmartly.ABSmartlyVue, {
  sdkOptions: {
    endpoint: "https://your-company.absmartly.io/v1",
    apiKey: "YOUR-API-KEY",
    environment: "production",
    application: "website",
  },
  context: {
    units: { user_id: "user-12345" },
  },
  attributes: {
    user_agent: navigator.userAgent,
  },
});

app.mount("#app");

// In your component, use treatments and track goals:
//
// <template>
//   <div>
//     <treatment name="homepage_banner_experiment">
//       <template #A>  <!-- Variant A (control) -->
//         <banner text="Welcome back!" />
//       </template>
//       <template #B>  <!-- Variant B -->
//         <banner text="Welcome back, we have new deals for you!" />
//       </template>
//     </treatment>
//
//     <button @click="trackClick">Get Started</button>
//   </div>
// </template>
//
// <script>
// export default {
//   methods: {
//     trackClick() {
//       this.$absmartly.track("cta_click", { page: "homepage" });
//     },
//   },
//   beforeUnmount() {
//     this.$absmartly.finalize();
//   },
// };
// </script>
